dbms/src/DataStreams/IProfilingBlockInputStream.cpp

```diff
@@ -110,36 +116,52 @@ Block IProfilingBlockInputStream::read(FilterPtr & res_filter, bool return_filte
     }
 #endif
 
-    info.updateExecutionTime(info.total_stopwatch.elapsed() - start_time);
+    {
+        auto lock = get_lock();
+        info.updateExecutionTime(info.total_stopwatch.elapsed() - start_time);
+    }
```

Comment:
In `progressImpl` function, `info` still may be used without lock protection.

Reply:
ditto